sky/services/platform/src/org/domokit/platform/SystemChromeImpl.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.domokit.platform;
+
+import android.app.Activity;
+import android.content.pm.ActivityInfo;
+import android.view.View;
+
+import org.chromium.mojo.system.MojoException;
+import org.chromium.mojom.flutter.platform.DeviceOrientation;
+import org.chromium.mojom.flutter.platform.SystemChrome;
+import org.chromium.mojom.flutter.platform.SystemUiOverlay
+;
+import org.domokit.activity.ActivityImpl;
+
+/**
+ * Android implementation of SystemChrome.
+ */
+public class SystemChromeImpl implements SystemChrome {
+    @Override
+    public void close() {}
+
+    @Override
+    public void onConnectionError(MojoException e) {}
+
+    @Override
+    public void setPreferredOrientations(int deviceOrientationMask,
+                                         SetPreferredOrientationsResponse callback) {
+        Activity activity = ActivityImpl.getCurrentActivity();
+        if (activity == null) {
+            callback.call(false);
+            return;
+        }
+
+        // Currently the Android implementation only supports masks with zero or one
+        // selected device orientations.
+        int androidOrientation;
+        if (deviceOrientationMask == 0) {
+            androidOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
+        } else if (deviceOrientationMask == DeviceOrientation.PORTRAIT_UP) {
```

Comment:
This is a mask. So something like `deviceOrientationMask & DeviceOrientation.PORTRAIT_UP != 0` perhaps